lib/Labyrinth/Players.hs

```diff
@@ -15,23 +16,40 @@ module Labyrinth.Players
 import           Prelude hiding (lookup)
 import           Lens.Micro     ((^.))
 import           Lens.Micro.TH  (makeLenses)
-import qualified Data.Map as M
+import           Data.Map       (Map)
+import qualified Data.Map       as M
 
 type Name = String
-data Color = Yellow | Blue | Green | Red deriving (Show, Eq, Ord)
+type PlayerStore = Map Color Player
+data Color = Clear | Yellow | Blue | Green | Red deriving (Show, Eq, Ord)
 
-data Players = Players (M.Map Color Player) deriving (Show, Eq)
-
-instance Monoid Players where
-  Players l `mappend` Players r = Players (M.union r l)
-  mempty = Players M.empty
-
-data Player = Player
+data Player = PlayerRemoved | Player
     { _color :: Color
     , _name  :: Name
     } deriving (Show, Eq)
 makeLenses ''Player
 
+data Players = Players PlayerStore deriving (Show, Eq)
+
+instance Monoid Color where
+  _ `mappend` c = c
+  mempty = Clear
+
+instance Monoid Players where
+  mempty = Players M.empty
+  Players l `mappend` Players r = Players (mergePlayers l r)
+
```

Comment:
@mamapitufo this is your idea for "tagging" the inverse: we take every player in `Players` and tag them as `PlayerRemoved`